pkg/apis/kappctrl/v1alpha1/types.go

```diff
@@ -77,8 +77,9 @@ type AppStatus struct {
 	ObservedGeneration int64          `json:"observedGeneration"`
 	Conditions         []AppCondition `json:"conditions"`
 
-	FriendlyDescription string `json:"friendlyDescription"`
-	UsefulErrorMessage  string `json:"usefulErrorMessage"`
+	FriendlyDescription  string `json:"friendlyDescription"`
+	UsefulErrorMessage   string `json:"usefulErrorMessage"`
+	HasDeployedResources bool
```

Comment:
Maybe it would it make sense to have something similar to `RefTracker` or `AppUpdateStatus` for Secret/ConfigMap updates? And then we could remove this from the status entirely.

Reply:
> for Secret/ConfigMap updates

lost you there. what would this do?

Reply:
Less about Secret/ConfigMap updates and more about how their state is stored (i.e. in memory). We could have Apps register that they have deployed resources in a map that can be accessed to check whether to proceed with delete in this case. We could also then remove `HasDeployedResources` from the status.

Reply:
this state though is not as transient as other state, since if you update kapp-controller you wouldnt know if something was previously deployed or not.